stripe/resource.py

```diff
@@ -393,7 +393,9 @@ def list(cls, api_key=None, idempotency_key=None,
         requestor = api_requestor.APIRequestor(api_key, account=stripe_account)
         url = cls.class_url()
         response, api_key = requestor.request('get', url, params)
-        return convert_to_stripe_object(response, api_key, stripe_account)
+        stripe_object = convert_to_stripe_object(response, api_key, stripe_account)
+        stripe_object._retrieve_params = params
```

Comment:
Yeah, I was wondering about this one too.

Here's what PEP-8 has to say about the leading underscore:

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose name starts with an underscore.

In this case, `_retrieve_params` should be considered a property of `StripeObject`. Because `ListObject` (a descendant of `StripeObject`) already accesses it, we could consider its visibility to be currently "protected" (based on the definition from other OO languages). `ListableAPIResource` is also a descendant of `StripeObject` so we're not really modifying that visibility from what it was before at all.

The idea of protected visibility is also in-line with PEP-8's idea of internal use, so I think we're okay on this here.